omx/gstomxtvp.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2018 RidgeRun LLC
+ *
+ * Author: Jose Lopez <jose.lopez@ridgerun.com>
+ *
+ * Based on gstomx_tvp.c made by:
+ *       David Soto <david.soto@ridgerun.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <gst/gst.h>
+#include <gst/base/base.h>
+#include <gst/controller/controller.h>
+
+#include <omx_vfcc.h>
+#include <OMX_TI_Index.h>
+#include <omx_ctrl.h>
+#include "gstomxtvp.h"
+#include "OMX_TI_Common.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_omxtvp_debug);
+#define GST_CAT_DEFAULT gst_omxtvp_debug
+
+enum
+{
+  PROP_0,
+  PROP_STANDARD,
+  PROP_SCAN_TYPE,
+};
+
+static GstStaticPadTemplate sink_template = GST_STATIC_PAD_TEMPLATE ("sink",
+    GST_PAD_SINK,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS ("ANY")
+    );
+
+static GstStaticPadTemplate src_template = GST_STATIC_PAD_TEMPLATE ("src",
+    GST_PAD_SRC,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS ("ANY")
+    );
+
+#define gst_omxtvp_parent_class parent_class
+G_DEFINE_TYPE (GstOMXTvp, gst_omxtvp, GST_TYPE_BASE_TRANSFORM);
+
+/* prototypes */
+static void gst_omx_tvp_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_omx_tvp_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+static gboolean gst_omx_tvp_start (GstBaseTransform * base);
+static GstFlowReturn gst_omx_tvp_transform_ip (GstBaseTransform * base,
+    GstBuffer * outbuf);
+static gboolean gst_omx_tvp_set_caps (GstBaseTransform * base,
+    GstCaps * incaps, GstCaps * outcaps);
+static gboolean gst_omx_tvp_stop (GstBaseTransform * base);
+static gboolean gst_omx_tvp_configure (GstOMXTvp * self);
+
+/* GObject vmethod implementations */
+
+/* tvp class init */
+static void
+gst_omxtvp_class_init (GstOMXTvpClass * klass)
+{
+  GObjectClass *gobject_class;
+  GstElementClass *gstelement_class;
+
+  gobject_class = (GObjectClass *) klass;
+  gstelement_class = (GstElementClass *) klass;
+
+  gobject_class->set_property = gst_omx_tvp_set_property;
+  gobject_class->get_property = gst_omx_tvp_get_property;
+
+  g_object_class_install_property (gobject_class, PROP_STANDARD,
+      g_param_spec_uint ("standard", "Video standard",
```

Comment:
No, this is not right. The user may set 989 if desired. You should create an enum